src/utils/dimensionality_operations.jl

```diff
@@ -1,8 +1,37 @@
+
 # Dimensionality operations for 4D Tensors
 # Author: Philipp Witte, pwitte3@gatech.edu
 # Date: January 2020
 
-export squeeze, unsqueeze, wavelet_squeeze, wavelet_unsqueeze, Haar_squeeze, invHaar_unsqueeze, tensor_split, tensor_cat
+export general_squeeze, general_unsqueeze, squeeze, unsqueeze, wavelet_squeeze, wavelet_unsqueeze, Haar_squeeze, invHaar_unsqueeze, tensor_split, tensor_cat
+
+####################################################################################################
+# General Squeeze and unsqueeze for user selection
+function general_squeeze(X::AbstractArray{T, N}; squeeze_type="normal", pattern="column") where {T, N}
```

Comment:
That's a poor name, the whole point of julia is multiple dispatch

```
squeeze(X::AbstractArray, type::String)
```
and dispatch on type. `normal` not a good name neither.

Reply:
Okay so something along the lines of changing everything to squeeze so:
`function squeeze(X::AbstractArray{T, N}; pattern="column") where {T, N}
`
would become 
`function squeeze(X::AbstractArray{T, N}, type::String="shuffle"; pattern="column") where {T, N}
`
and 
`function wavelet_squeeze(X::AbstractArray{T, N}; type=WT.db1) where {T, N}`
would become
`function squeeze(X::AbstractArray{T, N}, type::String="wavelet"; wavelet_type=WT.db1) where {T, N}`

etc etc?

Reply:
Im going to leave this change to another PR.